v1/services/storage/store.go

```diff
@@ -650,27 +648,145 @@ func (s *Store) ReadSeriesCardinality(ctx context.Context, req *datatypes.ReadSe
 		return nil, err
 	}
 
-	if req.Range.Start == 0 {
-		req.Range.Start = models.MinNanoTime
+	db, rp, start, end, err := s.validateArgs(source.OrgID, source.BucketID, req.Range.Start, req.Range.End)
+	if err != nil {
+		return nil, err
+	}
+
+	sgs, err := s.MetaClient.ShardGroupsByTimeRange(db, rp, time.Unix(0, start), time.Unix(0, end))
+	if err != nil {
+		return nil, err
+	}
+
+	if len(sgs) == 0 {
+		return cursors.NewInt64SliceIterator([]int64{0}), nil
+	}
+
+	var expr influxql.Expr
+	if root := req.Predicate.GetRoot(); root != nil {
+		expr, err = reads.NodeToExpr(root, measurementRemap)
+		if err != nil {
+			return nil, err
+		}
+
+		if found := reads.HasFieldValueKey(expr); found {
+			return nil, errors.New("field values unsupported")
+		}
+		expr = influxql.Reduce(influxql.CloneExpr(expr), nil)
+		if reads.IsTrueBooleanLiteral(expr) {
```

Comment:
Can you add a comment above this saying if it's an optimization or if it's needed to avoid a bug downstream? It looks like the sort of thing that somebody might "clean up" later if they read it without context.

Reply:
It is needed to allow the cursor to be created; the cursor doesn't like boolean literals. I also found that I need to handle a `false` boolean literal properly so I added support for that too.